2020-02-Spring/3_trees/README.md

```diff
@@ -116,21 +137,66 @@ Go to [Top](#top)
 <!-- Don't remove -->
 <a name="s2"/>
 
-### 2. SOLUTION 2 TODO :bug:
+### 2. Deepest Leaves Sum
 
-Source: TODO :bug:
-
-#### Naive/Simple Solution
-
-TODO :bug:
 
 #### Optimal Solution
 
-TODO :bug:
+<details>
+<summary>Click to see solution</summary>
+
+We need the sum of the leaves at the very very last level of the tree. To do this, we 
+can implement a Breath-first search traversal, tracking the sum of the values of the nodes at each level 
+of the tree, and reseting the sum everytime we go to a further level down the tree. .
+Once we get to the last level, we will get the sum of the value of the nodes at that level, exit the 
+loop, and return the sum.
+
+**C++:**
+```c++
+int deepestLeavesSum(TreeNode* root) {
+        
+        queue<TreeNode*> queue;
+        queue.push(root); // we insert the root into the queue
+        int currSum = 0;  //will store the sum of the nodes at the current level
+        
+        while(!queue.empty()){
+            currSum = 0; // reset the sum at each new level of the tree
+            int size = queue.size(); // locking the size of the queue at the current level
+									//before we add more nodes to it
+            
+            //for the size of the queue
+            for(int i =0; i< size;i++){
+                auto node = queue.front();
+                // Pop one node of the queue
+                queue.pop();
+                
+                currSum += node->val;
+                 //get left child into queue
+                if(node->left !=nullptr){
+                
+                    queue.push(node->left);
+                }
+                //get right child into queue
+                if(node->right !=nullptr){
+                    queue.push(node->right);
+                }
+               
+            }   
+        }
+        return currSum;   
+    }
+```c++
+	
+**Complexity Analysis:**
+- Space Complexity: O(n), because we are using a queue
```

Comment:
IMO, it's a little more readable to use math notation with monospace since it stands out. Feel free to ignore!